### ISSUE TITLE
merge LocalArray and LocalArrayRead, rename to LocalVector

### ISSUE THREAD

**Issue by stevengj** [JuliaParallel/PETSc.jl#77]
We could probably have a single type `LocalArray{T,readonly}` where `readonly` is `true` or `false`

Also, in keeping with Julia nomenclature, it should probably be called `LocalVector` since it is a subtype of `AbstractVector` (or `DenseVector` ala #76).

**Comment by JaredCrean2** [comment]
I like the name change.  The problem with making `readonly` a static parameter is I don't see a way to doing it that doesn't lead to the #59 or requires excessive information.  We could either have the outer constructor
```julia
LocalVector{T}(v::Vec; readonly=false)
  return LocalVector{t, readonly}(...args...)
```
which is type unstable, or the inner constructor:
```julia
type LocalVector{T, readonly}
  ... fields...
  function LocalVector(x::Vec)
     return new(...args...)
  end
end
```

which requires specify the element type `T` again, even though it is already contained in `x`.

**Comment by stevengj** [comment]
We could have type-stable outer constructors `LocalVector{T}(v::Vec{T}) = LocalVector{T,false}(...)` and `LocalVector_readonly{T}(v::Vec{T}) = LocalVector{T,true}(...)`

**Comment by JaredCrean2** [comment]
I'm not happy with the latter  name either, but I can't think of anything better, so let's go with it for now.

**Comment by JaredCrean2** [comment]
This was closed by #80
